changelogs/unreleased/bugfix-resource-action-cleanup.yml

```diff
@@ -0,0 +1,6 @@
+---
+description: Fix bug where the cleanup job, that removes old resource actions, ignores the environment scope of the `resource_action_logs_retention` setting. This way the most restrictive value of the `resource_action_logs_retention` setting across all environments was applied on all environments.
```

Comment:
I find "restrictive" to be slightly ambiguous here. It could be interpreted as a restriction on "compiles should be kept for x hours" but I think you mean "compiles should be cleaned up after x hours". Perhaps "most restrictive value" -> "shortest interval"?